### PR TITLE
Update branch and compare commits

### PR DESCRIPTION
**Pull request by reimai**
Adding apis to compare commits and update pull requests's branch from the base. 

The update-branch call is an [experimental api](https://docs.github.com/en/rest/overview/api-previews#update-a-pull-request-branch), so I've added an extra boolean parameter 'iAgreeToUseExperimentalApi' to match the api's usage of a specific header.

### CODE REVIEW

File: github4s/src/main/scala/github4s/interpreters/PullRequestsInterpreter.scala [hunk 1]
```diff
@@ -160,4 +160,19 @@ class PullRequestsInterpreter[F[_]](implicit client: HttpClient[F]) extends Pull
...
+      owner: String,
+      repo: String,
+      pullRequest: Int,
+      iAgreeToUseExperimentalApi: Boolean,
+      expectedHeadSha: Option[String] = None,
+      headers: Map[String, String] = Map()
+  ): F[GHResponse[BranchUpdateResponse]] =
+    client.put[BranchUpdateRequest, BranchUpdateResponse](
+      s"repos/$owner/$repo/pulls/$pullRequest/update-branch",
+      headers ++ Option("Accept" -> "application/vnd.github.lydian-preview+json")
```
Comment (BenFradet) [inline github4s/src/main/scala/github4s/interpreters/PullRequestsInterpreter.scala:174]: what happens if this header is not present?

**Review by BenFradet — COMMENTED** [review]
I have a question regarding the header before I delve into the PR completely

File: github4s/src/main/scala/github4s/interpreters/PullRequestsInterpreter.scala (hunk 1 above)
Comment (reimai) [reply github4s/src/main/scala/github4s/interpreters/PullRequestsInterpreter.scala:174]: In this case an [unsupported media type is returned](https://docs.github.com/en/rest/reference/pulls#update-a-pull-request-branch): ```Status: 415 Unsupported Media Type```

Comment (BenFradet) [reply github4s/src/main/scala/github4s/interpreters/PullRequestsInterpreter.scala:174]: Then I think we need to remove the `iAgree...` and document it's experimental in the scaladoc and example, wdyt?

Comment (reimai) [reply github4s/src/main/scala/github4s/interpreters/PullRequestsInterpreter.scala]: Sure, that would be less confusing for users

Comment (reimai) [reply github4s/src/main/scala/github4s/interpreters/PullRequestsInterpreter.scala]: Would you look at the changes?

File: github4s/src/main/scala/github4s/algebras/Repositories.scala [hunk 2]
```diff
@@ -228,6 +229,22 @@ trait Repositories[F[_]] {
       headers: Map[String, String] = Map()
   ): F[GHResponse[List[Commit]]]
 
+  /**
```
Comment (BenFradet) [inline github4s/src/main/scala/github4s/algebras/Repositories.scala:232]: maybe add what it does

**Review by BenFradet — COMMENTED** [review]
Looks good :+1: , is there a way to add an integration test for update branch?

File: github4s/src/main/scala/github4s/algebras/Repositories.scala (hunk 2 above)
Comment (reimai) [reply github4s/src/main/scala/github4s/algebras/Repositories.scala:232]: sure

**Review by juanpedromoreno — APPROVED** [review]
LGTM, thanks for your time @reimai !

**Pull request merged**